server/src/utils/mixerConnections/AtemConnection.ts

```diff
@@ -0,0 +1,415 @@
+import { Atem, AtemState } from 'atem-connection'
+
+import { store, state } from '../../reducers/store'
+import {
+    fxParamsList,
+    IMixerProtocol,
+} from '../../../../shared/src/constants/MixerProtocolInterface'
+import { logger } from '../logger'
+import { storeSetMixerOnline } from '../../../../shared/src/actions/settingsActions'
+import { IFader } from '../../../../shared/src/reducers/fadersReducer'
+import { IChannel } from '../../../../shared/src/reducers/channelsReducer'
+import { dbToFloat, floatToDB } from './LawoRubyConnection'
+import {
+    AudioMixOption,
+    FairlightAudioMixOption,
+    FairlightInputConfiguration,
+} from 'atem-connection/dist/enums'
+import {
+    storeFaderLevel,
+    storeInputGain,
+    storeSetMute,
+    storeSetPgm,
+    storeSetVo,
+} from '../../../../shared/src/actions/faderActions'
+import { storeSetChLabel } from '../../../../shared/src/actions/channelActions'
+import { FairlightAudioSource } from 'atem-connection/dist/state/fairlight'
+
+export class AtemMixerConnection {
+    private _connection: Atem
+
+    private _chNoToSource: Record<number, number> = {}
+    private _sourceToChNo: Record<number, number> = {}
+    private _sourceTracks: Record<number, '-65280' | '-255' | '-256'> = {}
+    private _firstConnection = true
+
+    constructor(
+        private mixerProtocol: IMixerProtocol,
+        public mixerIndex: number
+    ) {
+        this._connection = new Atem()
+
+        this._connection.on('connected', () => {
+            logger.info('Atem connected')
+            store.dispatch(storeSetMixerOnline(this.mixerIndex, true))
+            global.mainThreadHandler.updateMixerOnline(this.mixerIndex)
+
+            this.setupMixerConnection()
+        })
+        this._connection.on('disconnected', () => {
+            logger.info('Atem disconnected')
+            store.dispatch(storeSetMixerOnline(this.mixerIndex, false))
+            global.mainThreadHandler.updateMixerOnline(this.mixerIndex)
+        })
+
+        this._connection
+            .connect(state.settings[0].mixers[this.mixerIndex].deviceIp)
+            .catch(() => {
+                logger.error(
+                    'Failed to connect to atem ' +
+                        state.settings[0].mixers[this.mixerIndex].deviceIp
+                )
+            })
+    }
+
+    setupMixerConnection() {
+        const sourceToName: Record<string, string> = {
+            ...Object.fromEntries(
+                Object.entries(this._connection.state.inputs).map(
+                    ([id, input]) => [id, input.shortName]
+                )
+            ),
+            '1301': 'Mic 1',
+            '1302': 'Mic 2',
```

Comment:
Why are these different than the ones from the connection state inputs?

Reply:
They are not included in that object with inputs. I think it's because they are not video sources. The atem software also doesn't let you set names for them, they seem to be hardcoded to these mic 1/mic 2 names in their software.